lfs/pointer.go

```diff
@@ -109,16 +109,28 @@ func DecodePointer(reader io.Reader) (*Pointer, error) {
 }
 
 func DecodeFrom(reader io.Reader) ([]byte, *Pointer, error) {
+	output, _, p, err := DecodeFromHasMore(reader)
+	return output, p, err
+}
+
+// DecodeFromHasMore decodes an *lfs.Pointer from the given io.Reader, "reader".
+// If the pointer encoded in the reader could successfully be read and decoded,
+// it will be returned with a nil error.
+//
+// If the pointer could not be decoded, a buffer of the data read so far, along
+// with whether or not there was more data not yet consumed will be returned
+// instead.
+func DecodeFromHasMore(reader io.Reader) ([]byte, bool, *Pointer, error) {
```

Comment:
Just for posterity, I mentioned in video chat that the `DecodeFromHasMore` return signature feels clunky. My suggestion was to change the `DecodeFrom` interface to `(io.Reader, *Pointer, error)`, and let the caller just drain the returned reader if it wants. Callers shouldn't have to worry about how to join the read buffer with the io reader.